pkg/interfaces/contracts/pool-stable/StablePoolUserData.sol

```diff
@@ -15,8 +15,8 @@
 pragma solidity ^0.7.0;
 
 library StablePoolUserData {
-    enum JoinKind { INIT, EXACT_TOKENS_IN_FOR_BPT_OUT, TOKEN_IN_FOR_EXACT_BPT_OUT }
-    enum ExitKind { EXACT_BPT_IN_FOR_ONE_TOKEN_OUT, BPT_IN_FOR_EXACT_TOKENS_OUT }
+    enum JoinKind { INIT, EXACT_TOKENS_IN_FOR_BPT_OUT, TOKEN_IN_FOR_EXACT_BPT_OUT, ALL_TOKENS_IN_FOR_EXACT_BPT_OUT }
+    enum ExitKind { EXACT_BPT_IN_FOR_ONE_TOKEN_OUT, EXACT_BPT_IN_FOR_ALL_TOKENS_OUT, BPT_IN_FOR_EXACT_TOKENS_OUT }
```

Comment:
Hey guys! First of all thanks so much for adding these operations back in, was in the process of doing it myself and preparing to beg when I saw this PR 😅.

Adding this exit option in the middle here aligns with the previous config, but will now break backwards compatibility with the exiting composable stable. It's a trade off, but I would say that it might be better to not break backwards compatibility as it increases implementation complexity.

Reply:
I had it that way, but flipped it because of https://github.com/balancer-labs/balancer-v2-monorepo/pull/1912#discussion_r1006137221. Sounds like I should change it back then. @nventuro ?

Reply:
@danielmkm what approach do you think would be better? If done as in this PR, we only have two variants: those with and those without proportional exits (same as we have today).

Reply:
I think the best approach would be to preserve the existing ComposableStable ordering, so swap `BPT_IN_FOR_EXACT_TOKENS_OUT` and `EXACT_BPT_IN_FOR_ALL_TOKENS_OUT` in the enum definition.

```
enum ExitKind { EXACT_BPT_IN_FOR_ONE_TOKEN_OUT, BPT_IN_FOR_EXACT_TOKENS_OUT, EXACT_BPT_IN_FOR_ALL_TOKENS_OUT}
```

By doing this, any integration that gets built against the current ComposableStable will be able to support the new pool version without making any code changes. They can then choose to add proportional join/exit support on their own timeline.

With the PR as it is, the enum reordering is a breaking change. So any existing integration cannot just use the new version without also making code changes, usually meaning a longer timeline for adoption as additional systems need to be updated to account for the changes.

When doing the first integration I lost some time not realizing the enum value of `BPT_IN_FOR_EXACT_TOKENS_OUT` had changed, and the error code you get is the dreaded 430 😅.